File: mobile/lib/models/job_offer.dart
class JobOffer {
  final int? id;
  final String title;
  final String description;
  final String? location;
  final String? contractType;
  final String? requiredSkills;
  final int? experienceYears;
  final String? salaryRange;
  final String? status;
  final int? departmentId;
  final String? deadline;

  const JobOffer({
    this.id,
    required this.title,
    required this.description,
    this.location,
    this.contractType,
    this.requiredSkills,
    this.experienceYears,
    this.salaryRange,
    this.status,
    this.departmentId,
    this.deadline,
  });

  factory JobOffer.fromJson(Map<String, dynamic> json) {
    return JobOffer(
      id: json['id'] != null ? (json['id'] as num).toInt() : null,
      title: json['title']?.toString() ?? '',
      description: json['description']?.toString() ?? '',
      location: json['location']?.toString(),
      contractType: json['contractType']?.toString(),
      requiredSkills: json['requiredSkills']?.toString(),
      experienceYears: json['experienceYears'] != null
          ? (json['experienceYears'] as num).toInt()
          : null,
      salaryRange: json['salaryRange']?.toString(),
      status: json['status']?.toString(),
      departmentId: json['departmentId'] != null
          ? (json['departmentId'] as num).toInt()
          : null,
      deadline: json['deadline']?.toString(),
    );
  }
}

File: backend/recruitment-service/src/main/java/com/example/recruitment/model/JobOffer.java
package com.example.recruitment.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "job_offers")
public class JobOffer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String location;
    private String contractType; // CDI, CDD, STAGE, FREELANCE

    @Column(name = "required_skills", columnDefinition = "TEXT")
    private String requiredSkills; // JSON array as string e.g. ["Java","Spring","SQL"]

    private Integer experienceYears;
    private String salaryRange; // e.g. "30000-40000 TND"

    @Enumerated(EnumType.STRING)
    private JobStatus status = JobStatus.OPEN;

    @Column(name = "created_by_user_id")
    private Long createdByUserId; // RH_MANAGER id

    @Column(name = "department_id")
    private Long departmentId;

    private LocalDate deadline; // application deadline
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
